src/cpp/fastdds/publisher/DataWriterImpl.cpp

```diff
@@ -464,25 +464,25 @@ ReturnCode_t DataWriterImpl::enable()
     topic_desc.type_name = topic_->get_type_name();
     publisher_->get_participant_impl()->fill_type_information(type_, topic_desc.type_information);
 
-    WriterQos wqos = qos_.get_writerqos(get_publisher()->get_qos(), topic_->get_qos());
-    if (!is_data_sharing_compatible_)
+    PublicationBuiltinTopicData publication_data;
+    if (get_publication_builtin_topic_data(publication_data) != RETCODE_OK)
     {
-        wqos.data_sharing.off();
+        EPROSIMA_LOG_ERROR(DATA_WRITER, "Error getting publication data. RTPS Writer not enabled.");
+        return RETCODE_ERROR;
     }
-    if (endpoint_partitions)
+    RTPSParticipantImpl* rtps = RTPSDomainImpl::find_local_participant(publisher_->rtps_participant()->getGuid());
+    ReturnCode_t register_writer_code = rtps->register_writer(writer_, topic_desc, publication_data);
+    if (register_writer_code != RETCODE_OK)
     {
-        std::istringstream partition_string(*endpoint_partitions);
-        std::string partition_name;
-        wqos.m_partition.clear();
-
-        while (std::getline(partition_string, partition_name, ';'))
-        {
-            wqos.m_partition.push_back(partition_name.c_str());
-        }
```

Comment:
I think I dont fully understand why we are removing the partitions checks from here